bitcoin/src/address/mod.rs

```diff
@@ -289,14 +289,17 @@ pub enum AddressData {
 /// // variant 1
 /// let address: Address<NetworkUnchecked> = "32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf".parse().unwrap();
 /// let address: Address<NetworkChecked> = address.require_network(Network::Bitcoin).unwrap();
+/// assert_eq!(address.to_string(), "32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf");
 ///
 /// // variant 2
 /// let address: Address = Address::from_str("32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf").unwrap()
 ///                .require_network(Network::Bitcoin).unwrap();
+/// assert_eq!(address.to_string(), "32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf");
 ///
 /// // variant 3
 /// let address: Address<NetworkChecked> = "32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf".parse::<Address<_>>()
 ///                .unwrap().require_network(Network::Bitcoin).unwrap();
+/// assert_eq!(address.to_string(), "32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf");
```

Comment:
I think these three are unnecessary. My reasoning is that rustdoc examples are examples to read but also an initial unit test, so functionally this is three roundtrip tests with the same input value. I personally would prefer to see
```rust
/// let _: Address<NetworkChecked> = address.require_network(Network::Bitcoin).unwrap();
```

Reply:
The point is to demonstrate that `.to_string()` produces the text address that somebody would pass around.

Do you think that's too obvious?

Reply:
After looking at this again just now, I agree that the first 3 `assert_eq!` in the variant section are unnecessary, since just below there is another section `### Formatting addresses` where it makes more sense to give an example of using `.to_string()` in the `assert_eq!`.

I will leave changing it though until I get some feedback since I also just added two patches to this PR.

Reply:
I agree with @apoelstra that there is merit in showing what `to_string` does and also with @jamillambert that its best left to the formatting section below. Which implies I still agree with myself from days ago but for additional reasons, which is nice :)

Reply:
I have removed the first 3 `to_string` and instead prefixed the variables with a `_`, to match what I did elsewhere.